### PR TITLE
fix: some function should pass context parameter

### PR DESCRIPTION
**Pull request by Abirdcfly**
some function use non-inherited context

#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes # 

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go [hunk 1]
```diff
@@ -159,7 +159,7 @@ func (pl *VolumeRestrictions) isReadWriteOncePodAccessModeConflict(pod *v1.Pod)
 		pvcKeys = append(pvcKeys, key)
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
+	subCtx, cancel := context.WithCancel(ctx)
```
Comment (damemi) [inline pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:162]: do you still need to create a context in here or can you just use the one that's passed to the function now?

File: pkg/scheduler/framework/plugins/interpodaffinity/filtering.go [hunk 2]
```diff
@@ -115,7 +115,8 @@ func (m topologyToMatchedTermCount) update(node *v1.Node, tk string, value int64
 // updates the topologyToMatchedTermCount map with the specified value
 // for each affinity term if "targetPod" matches ALL terms.
 func (m topologyToMatchedTermCount) updateWithAffinityTerms(
-	terms []framework.AffinityTerm, pod *v1.Pod, node *v1.Node, value int64) {
```
[the remark this answers is not in the document]
Comment (Abirdcfly) [reply pkg/scheduler/framework/plugins/interpodaffinity/filtering.go]: Update. And this line will be auto format just use gofmt....😂

File: pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go (hunk 1 above)
Comment (Abirdcfly) [reply pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:162]: the sub-context `subCtx` try to cancel the function when "node has pod using PersistentVolumeClaim with the same name and ReadWriteOncePod access mode", and `ctx` from the function args try to cancel the function When the external cancel function is triggered. I think they are different.

Comment (kerthcet) [reply pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:162]: So if we want to cancel the `PreFilter` phase, why we still want to run `processNode`? I think this is the concern Mike cares?

Comment (Abirdcfly) [reply pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:162]: @denkensk @kerthcet  `subCtx` is created by `context.WithCancel`, so when parent context(`ctx` in function args)'s cancel function is called, the `processNode` function will cancel. This process (new) is working fine and this is what I added in this pr. But here is another process (old), concurrent execution of multiple `processNode ` functions, if a certain one does not meet the condition`refCount > 0`, then all running `processNode ` functions are cancelled, by `atomic.AddUint32(&conflicts, 1); cancel()`. Similar process appears in the k8s code [scheduler.preemption](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/preemption/preemption.go#L551)

Comment (kerthcet) [reply pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:162]: Sorry for the rough first review, this makes sense as we may cancel the parallel program with `cancel()`. Thanks for explaining.

**Comment by Abirdcfly** [comment]
kindly ping @ahg-g @damemi PTAL

**Pull request merged**